### ISSUE TITLE
Generated BabelNodeClassMethod type doesn't include `static` from babel-types

### ISSUE THREAD

**Issue by chicoxyzzy** [babel/babel#6203]
Generated [`BabelNodeClassMethod`](https://github.com/babel/babel/blob/master/lib/types.js#L473-L489) doesn't include [`static`](https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/es2015.js#L378)  field from babel-type.

This leads other libraries to [create workarounds](https://github.com/facebook/prepack/issues/661)

| software         | version(s)
| ---------------- | -------
| Babel            | 6+
| node             | any
| npm              | any
| Operating System | any

**Comment by hzoo** [comment]
We don't really want to do backports anymore? Those generated types are really buggy anyway, it should probably get types from babylon instead which has flow support. https://github.com/babel/babylon/blob/master/src/types.js

**Comment by chicoxyzzy** [comment]
Should completely remove `/lib/types.js` in Babel 7 than?

**Comment by hzoo** [comment]
Yeah maybe actually, another reason for us to move babylon into the monorepo because it would allow us to reuse and test that those types are correct everywhere.

**Comment by chicoxyzzy** [comment]
> it should probably get types from babylon instead which has flow support

Than we should publish libdefs for babylon or just use [`flow-copy-source`](https://github.com/AgentME/flow-copy-source).

**Comment by chicoxyzzy** [comment]
I think we should fix this or remove `babel-types` and use `babel-plugin-transform-flow-comments` in babylon before 7 release. If we decide to remove `babel-types` I can do this and update all dependent packages in the repo after moving babylon to monorepo.
